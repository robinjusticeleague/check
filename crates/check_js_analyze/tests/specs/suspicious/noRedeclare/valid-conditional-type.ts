/* should not generate diagnostics */
// Issue https://github.com/checkjs/check/issues/2659
type Test<T> = T extends Array<infer U> ? true : false