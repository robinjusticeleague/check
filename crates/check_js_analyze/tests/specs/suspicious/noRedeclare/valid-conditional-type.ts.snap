---
source: crates/check_js_analyze/tests/spec_tests.rs
expression: valid-conditional-type.ts
---
# Input
```ts
/* should not generate diagnostics */
// Issue https://github.com/checkjs/check/issues/2659
type Test<T> = T extends Array<infer U> ? true : false
```
